discovery-infra/test_infra/junit_report/junit_test_suite.py

```diff
@@ -0,0 +1,54 @@
+import os
+from typing import Callable, Union, Type
+
+import decorator
+from junit_xml import TestSuite, to_xml_report_string
+
+from .junit_test_case import JunitTestCase
+
+
+class JunitTestSuite:
+    _report_dir: str
+    _func: Union[Callable, None]
+    _suite: Union[TestSuite, None]
+    _name = str
+
+    def __init__(self, report_dir="./"):
```

Comment:
`os.path.curdir`

Reply:
even better `pathlib.Path()` then you can just `self._report_dir / f"junit_{self._suite.name}_results.xml"` instead of `os.path.join`. The type hint is also more clear because it becomes `_report_dir: pathlib.Path` rather than `str`